### DISCUSSION TITLE
OpenMC Wrapper Seg Fault

### DISCUSSION THREAD

**Discussion by folkthom** [neams-th-coe/cardinal#421] [Q&A] [answered]
Hi,

I have a case that I am trying to couple OpenMC to MOOSE heat conduction. I've been following the tutorials -- which have been very helpful -- but I am encountering a seg fault in the OpenMC run. This does not occur when I run OpenMC standalone; only with Cardinal.

I've attached the OpenMC file of the Cardinal run. I did not include the heat conduction file as it seems to work fine, and instead initialized the temperature in the OpenMC file.

Any insight into why I am receiving a seg fault would be much appreciated. Thanks!

[seg_fault.zip](https://github.com/neams-th-coe/cardinal/files/9096071/seg_fault.zip)

**Comment by aprilnovak** [comment, marked as answer]
Taking a look now! Would you be able to share the Python script you used to build the OpenMC model? 

It's not a _fix_, but setting `initial_properties = xml` on `OpenMCCellAverageProblem` will take the initial temperature from the XML files (in effect, just running a standalone OpenMC via the Cardinal executable). This eliminated your seg fault, which gives me a hint that it's something about the temperature IC in Cardinal.

Having the Python files will let me debug this further. Thanks :)

**Reply by aprilnovak** [reply]
This has been addressed in #423.
